### PR TITLE
bpo-36396: Remove fgBg param of idlelib.config.GetHighlight()

### PR DESCRIPTION
**Pull request by terryjreedy**
This param was only used twice and changed the return type.

https://bugs.python.org/issue36396

### CODE REVIEW

**Comment by terryjreedy** [comment]
Cheryl, what you you think of changing 'd' to 'hi', etc in the configdialog tests?  I only changed the one test I edited.

File: Lib/idlelib/config.py [hunk 1]
```diff
@@ -283,34 +283,20 @@ def GetSectionList(self, configSet, configType):
...
-        highlight = {"foreground": fore, "background": back}
-        if not fgBg:  # Return dict of both colors
-            return highlight
-        else:  # Return specified color only
-            if fgBg == 'fg':
-                return highlight["foreground"]
-            if fgBg == 'bg':
-                return highlight["background"]
-            else:
-                raise InvalidFgBg('Invalid fgBg specified')
```
Comment (csabella) [inline Lib/idlelib/config.py:313]: This is the only usage of `InvalidFgBg`, so I think the definition can be removed as well.

File: Lib/idlelib/NEWS.txt [hunk 2]
```diff
@@ -3,6 +3,11 @@ Released on 2019-10-20?
 ======================================
 
 
+bpo-36396: Remove fgBg param of idlelib.config.GetHighlight()
+This param was only used twice and changed the return type.
+
+bpo-23216: IDLE: Add docstrings to search modules
```
Comment (csabella) [inline Lib/idlelib/NEWS.txt]: nitpick: add a period at the end.

File: Misc/NEWS.d/next/IDLE/2019-03-21-22-43-21.bpo-36396.xSTX-I.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Remove fgBg param of idlelib.config.GetHighlight() This param was only used
```
Comment (csabella) [inline Misc/NEWS.d/next/IDLE/2019-03-21-22-43-21.bpo-36396.xSTX-I.rst]: Missing period or semicolon.

**Review by csabella — COMMENTED** [review]
I really like this change.  Some small things in other comments.  I probably would have used f-strings instead of concatenation, but that's just a personal preference.  :-)

File: Lib/idlelib/idle_test/test_configdialog.py [hunk 4]
```diff
@@ -606,40 +606,37 @@ def test_set_color_sample(self):
...
         eq = self.assertEqual
-        d = self.page
-        del d.paint_theme_sample
-        hs_tag = d.highlight_sample.tag_cget
+        hi = self.page
+        del hi.paint_theme_sample  # Delete masking mock.
+        hs_tag = hi.highlight_sample.tag_cget
         gh = idleConf.GetHighlight
         fg = 'foreground'
         bg = 'background'
```
Comment (csabella) [inline Lib/idlelib/idle_test/test_configdialog.py]: I don't think `fg` or `bg` is used anymore.

**Comment by csabella** [comment]
> Cheryl, what you you think of changing 'd' to 'hi', etc in the configdialog tests? I only changed the one test I edited.

So, `d` was for `dialog` and it's on all the tests.  It was a pretty clean replacement at the time to go from `d = dialog` to `d = self.page`, but the `d` doesn't really have a meaning anymore.  Since other recent tests have used `text = self.text`, if I was doing this now, I'd probably use `page = self.page`.  It's a little more readable than `d`.  However, if each page used a unique value, such as `hi`, then it would be easier to see immediately which page was being tested without needing to look back at the class.  I guess I'd be +0 on the change.  Once you look at the source and know what `d` is, it makes it easy to read all the tests because it's consistent.  But, there might be a slight cognitive gain from changing it.

File: Lib/idlelib/NEWS.txt (hunk 2 above)
Comment (terryjreedy) [reply Lib/idlelib/NEWS.txt]: Nitpicks wanted.  Both periods added.

File: Lib/idlelib/config.py (hunk 1 above)
Comment (terryjreedy) [reply Lib/idlelib/config.py:313]: Good catch. Gone.

File: Lib/idlelib/idle_test/test_configdialog.py (hunk 4 above)
Comment (terryjreedy) [reply Lib/idlelib/idle_test/test_configdialog.py]: Ditto.  My initial changes still used them, but in the end, the literals are only needed in one place, console_new, and I in-lined them there.  Gone.

File: Lib/idlelib/config.py (hunk 1 above)
Comment (terryjreedy) [reply Lib/idlelib/config.py:313]: Gone.

File: Misc/NEWS.d/next/IDLE/2019-03-21-22-43-21.bpo-36396.xSTX-I.rst (hunk 3 above)
Comment (terryjreedy) [reply Misc/NEWS.d/next/IDLE/2019-03-21-22-43-21.bpo-36396.xSTX-I.rst]: Added.

**Comment by terryjreedy** [comment]
Thanks for the fixes.  

While editing, I felt that 'd' was a nuisance because I kept mentally translating it each time I read it.  I thought of 'high', 'page'  or 'hp' instead of 'd'.  'hi' instead of 'high' was an experiment.  For the 5 test classes, 'font',  'dent', 'high', 'keys', and 'gen' could work, but don't feel completely right.  I think 'page' is what I was mentally substituting for 'd'.  Given your comment, I would go with that *and* improve some of the test names to be more page specific.  For example, change 'test_builtin_name' to 'test_builtin_theme' and 'test_builtin_keys' for the respective pages.  For now, I changed 'hi' to 'page' and intend to leave that for the merge.

**Comment by terryjreedy** [comment]
I did not see the f-string comment until now.  They would be fine and for anything other than `name + 'string'` I would change.  Even `f'{name1}{name2}'` has the advantage over name1 + name2 of clearly resulting in a string, with some green highlighting.

**Pull request merged**